tests/integration/container_test.py

```diff
@@ -263,8 +263,16 @@ def test_create_with_mac_address(self):
 
         self.client.start(container)
         res = self.client.inspect_container(container['Id'])
-        self.assertEqual(mac_address_expected,
-                         res['NetworkSettings']['MacAddress'])
+        try:
+            self.assertEqual(
+                mac_address_expected,
+                res['NetworkSettings']['Networks']['bridge']['MacAddress']
+            )
+        except KeyError:
```

Comment:
I think a comment about why this needs to exist would be great (so we don't forget). Something about which API version would hit which branch.

That way we know when we can remove it as well.